src/controllers/post.controller.js

```diff
@@ -24,58 +25,65 @@ const addNewPost = asyncHandler(async (req, res, next) => {
   res.status(201).json(new ApiResponse(201, { newPost }, "Post created"));
 });
 
-// ------------------------- Update Post -------------------------
+// ------------------------- Update self Post -------------------------
 const updatePost = asyncHandler(async (req, res, next) => {
-  const { postId, image, caption, likes, comments } = req.body;
+  const { postId, caption } = req.body;
+  const userIdFromToken = req.user._id;
 
-  if (!(caption && postId)) {
+  if (bodyDataExists(postId && caption)) {
```

Comment:
is this the condition that you are going for ?

Reply:
yes